File: caffeine/src/main/java/project/study/caffeine/core/web/city/presentation/response/DistrictResponse.java
package project.study.caffeine.core.web.city.presentation.response;

import project.study.caffeine.core.domain.city.entity.District;

public class DistrictResponse {

    private final Long districtId;
    private final String districtName;
    private final String cityName;

    private DistrictResponse(District district) {
        this.districtId = district.getDistrictId();
        this.districtName = district.getName();
        this.cityName = district.getCityName();
    }

    public static DistrictResponse of(District district) {
        return new DistrictResponse(district);
    }

    public Long getDistrictId() {
        return districtId;
    }

    public String getDistrictName() {
        return districtName;
    }

    public String getCityName() {
        return cityName;
    }
}

File: jpa/src/main/kotlin/project/study/jpa/core/web/city/presentation/response/DistrictResponse.kt
package project.study.jpa.core.web.city.presentation.response

import project.study.jpa.core.domain.city.entity.District

class DistrictResponse(district: District) {

    private val districtId: Long = district.getDistrictId()!!
    private val districtName: String = district.getName()
    private val cityName: String = district.getCityName()

    fun getDistrictId(): Long {
        return districtId
    }

    fun getDistrictName(): String {
        return districtName
    }

    fun getCityName(): String {
        return cityName
    }
}
